File: php/features.php
<?php
declare(strict_types=1);

// StarWarsDatabank SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class StarWarsDatabankFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new StarWarsDatabankBaseFeature();
            case "test":
                return new StarWarsDatabankTestFeature();
            default:
                return new StarWarsDatabankBaseFeature();
        }
    }
}

File: py/features.py
# StarWarsDatabank SDK feature factory

from feature.base_feature import StarWarsDatabankBaseFeature
from feature.test_feature import StarWarsDatabankTestFeature


def _make_feature(name):
    features = {
        "base": lambda: StarWarsDatabankBaseFeature(),
        "test": lambda: StarWarsDatabankTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
